src/redux/app/saga.js

```diff
@@ -148,17 +148,28 @@ function* initFromStorageRequest() {
 
 function* loginRequest(action) {
   const { isRelogin } = action;
-  const appId = application.get('id');
   try {
-    // Read Parse.User from storage, if not, sign in or sign up
-    const user = yield call(ParseHelper.getUser);
-    console.log('loginRequest, read from storage, user: ', user);
-    // If you need to log in again, or the user exists, call ParseHelper.signInOrSignUp
-    // Else use the user from storage
-    if (!user || isRelogin) {
-      yield call(ParseHelper.signInOrSignUp, appId);
+    const password = yield call(storage.getUserPassword);
+    // If the password does not exist, it means this is a new user, then sign up.
+    // Else login.
+    if (_.isEmpty(password)) {
+      yield call(ParseHelper.signUp);
+    } else {
+      // Read Parse.User from storage, if not, sign in or sign up
+      const user = yield call(ParseHelper.getUser);
+      console.log('loginRequest, read from storage, user: ', user);
+      // If you need to log in again, or the user exists, call ParseHelper.signInOrSignUp
+      // Else use the user from storage
+      if (!user || isRelogin) {
+        console.log('loginRequest, read from storage, user: ', user);
+        const appId = application.get('id');
+        yield call(ParseHelper.signInOrSignUp, appId, password);
```

Comment:
From this PR I see that you removed signInOrSignUp() function, no?

Reply:
Yes, I divided it into signIn and signUp